### PR TITLE
Dev s2 v2

### PR DESCRIPTION
**Pull request by Surfboy678**
In both cases only tests are passed setFitnessFullMatch.
Any ideas?

### CODE REVIEW

**Comment by PawelDabrowski83** [comment]
Please check issues #39 and #41 because they are crucial for understanding/fixing and passing the tests.

File: src/main/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImpl.java [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
-    @Override
-    public float setFitness(Gene gene) {
-        return 0;
-    }
+  public Variant1EvaluatorImpl(char target) {
+    this.target = target;
+  }
+  @Override
+  public float setFitness(Gene gene) {
+    float delta = gene.getValues()[0] - target;
```
Comment (PawelDabrowski83) [inline src/main/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImpl.java:16]: this is not stated directly in docs, but we assume that delta is never below zero, so Math.abs() will solve this

File: src/main/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImpl.java [hunk 2]
```diff
@@ -2,16 +2,18 @@
...
+    return (float) (1 / (1 + log(1 + delta)));
```
Comment (PawelDabrowski83) [inline src/main/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImpl.java:17]: tests use other logarithm function, hence they will not pass

File: src/main/java/com/facebook/genAlgo/evaluator/Variant2EvaluatorImpl.java [hunk 3]
```diff
@@ -4,8 +4,15 @@
...
+    private final char target;
+
+    public Variant2EvaluatorImpl(char target) {
+        this.target = target;
+    }
+
     @Override
     public float setFitness(Gene gene) {
-        return 0;
+        float delta = gene.getValues()[0] - target;
```
Comment (PawelDabrowski83) [inline src/main/java/com/facebook/genAlgo/evaluator/Variant2EvaluatorImpl.java:15]: Math.abs() as in Variant1 comment

File: src/main/java/com/facebook/genAlgo/evaluator/Variant2EvaluatorImpl.java [hunk 4]
```diff
@@ -4,8 +4,15 @@
...
+        return(float) ((655- delta) / 65535);
```
Comment (PawelDabrowski83) [inline src/main/java/com/facebook/genAlgo/evaluator/Variant2EvaluatorImpl.java:16]: value 655 is wrong, probably a typo

**Review by PawelDabrowski83 — CHANGES_REQUESTED** [review]

**Comment by Surfboy678** [comment]
In the second variant, it passes only setFitnessFullMatch.

**Review by PawelDabrowski83 — APPROVED** [review]
Now it passes all tests, but you should first update your local dev-s3 branch with latest bugfix and then merge it into your PR.

**Review by PawelDabrowski83 — APPROVED** [review]

**Pull request merged**